File: core/src/main/java/by/itechart/warehouse/dto/PlacementGoodsDto.java
package by.itechart.warehouse.dto;

import by.itechart.common.dto.GoodsDto;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.datatype.jsr310.deser.LocalDateDeserializer;
import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateSerializer;
import lombok.Data;

import java.time.LocalDate;

@Data
public class PlacementGoodsDto {
    private Long id;
    private Integer amount;
    private GoodsDto goods;
    private Long counterpartyId;
    private Integer storageTimeDays;
    @JsonSerialize(using = LocalDateSerializer.class)
    @JsonDeserialize(using = LocalDateDeserializer.class)
    private LocalDate expirationDate;
}

File: web-app/online-warehouse/src/app/warehouse/dto/placement-goods.dto.ts
import {GoodsDto} from "../../shared/goods/dto/goods.dto";

export class PlacementGoodsDto {
  id: number;
  amount: number;
  goods: GoodsDto;
  counterpartyId: number;
  storageTimeDays: number;
  expirationDate: Date;

  constructor(id?: number, amount?: number, goods?: GoodsDto, counterpartyId?: number, storageTimeDays?: number, expirationDate?: Date) {
    this.id = id;
    this.amount = amount;
    this.goods = goods;
    this.counterpartyId = counterpartyId;
    this.storageTimeDays = storageTimeDays;
    this.expirationDate = expirationDate;
  }
}
